File: implementAPI.cpp
vector<string> implementAPI(vector<string> &logs){
vector<string> ans;
map<string, pair<string, bool>> mp;
for(string&s : logs){
	vector<string> temp;
	stringstream str(s);
	string word;
	while(str >> word){
		temp.push_back(word);
	}
	string first = temp[0];
	string user = temp[1];
	if(first == "register"){
		string pass = temp[2];
		if(mp.find(user) != mp.end()){
			ans.push_back("Username already exists");
		}
		else{
			mp[user] = {pass, false};
			ans.push_back("Registered Successfully");
		}
	}
	else if(first == "login"){
		string pass = temp[2];
		if(mp.find(user) == mp.end()){
			ans.push_back("Login Unsuccessful");
		}
		else if(mp[user].first != pass || mp[user].second == true){
			ans.push_back("Login Unsuccessful");
		}
		else if(mp[user].first == pass){
			ans.push_back("Logged In Successfully");
			mp[user].second = true;
		}
	}
	else if(first == "logout"){
		if(mp.find(user) == mp.end() || mp[user].second == false){
			ans.push_back("Logout Unsuccessful");
		}
		else{
			ans.push_back("Logged Out Successfully");
			mp[user].second = false;
		}
	}
}
return ans;
}

File: implementAPI.java
public static List<String> implementAPI(List<String> logs) {

    Map<String, Boolean> loggedIn = new HashMap<>();
    Map<String, String> password = new HashMap<>();

    List<String> res = new ArrayList<>();

    for (String l : logs) {
        String[] temp = l.split(" ");
        if (temp[0].equals("register")) {
            if (password.containsKey(temp[1]))
                res.add("Username already exists");
            else {
                password.put(temp[1], temp[2]);
                res.add("Registered Successfully");
            }
        } else if (temp[0].equals("login")) {
            if (password.containsKey(temp[1]) && password.get(temp[1]).equals(temp[2])) {
                if (loggedIn.containsKey(temp[1])) {
                    if (loggedIn.get(temp[1]) == false) {
                        loggedIn.put(temp[1], true);
                        res.add("Logged In Successfully");
                    } else {
                        res.add("Login Unsuccessful");
                    }
                } else {
                    loggedIn.put(temp[1], true);
                    res.add("Logged In Successfully");
                }
            } else
                res.add("Login Unsuccessful");
        } else if (temp[0].equals("logout")) {
            if (password.containsKey(temp[1]) && loggedIn.containsKey(temp[1]) && loggedIn.get(temp[1]) == true) {
                loggedIn.put(temp[1], false);
                res.add("Logged Out Successfully");
            } else {
                res.add("Logout Unsuccessful");
            }
        }
    }

    return res;
}